index.ts

```diff
@@ -0,0 +1,148 @@
+import WeatherDisplay from "./components/WeatherDisplay.js";
+import TemperatureUnitToggle from "./components/TemperatureUnitToggle.js";
+import BackgroundChanger from "./components/BackgroundChanger.js";
+import TimeDisplay from "./components/TimeDisplay.js";
+import MapDisplay from "./components/MapDisplay.js";
+import WeatherIcons from "./components/WeatherIcons.js";
+import LanguageToggle from "./components/LanguageToggle.js";
+import Search from "./components/Search.js";
+
+interface Location {
+  latitude: number;
+  longitude: number;
+}
+
+interface WeatherData {
+  main: number;
+  feelsLike: number;
+  day1: number;
+  day2: number;
+  day3: number;
+}
+
+class WeatherApp {
+  private backgrounds: string[];
+  private sectionElement: Element | null;
+  private refreshButton: Element | null;
+  private location: Location;
```

Comment:
Default values could be defined in a different ways, one of them is using constructor. Another as properties.
https://learn.javascript.ru/class#svoystva-klassov
```suggestion
  private sectionElement: Element | null = document.querySelector(".section") ?? null;
  private refreshButton: Element | null = document.querySelector(".refresh-button") ?? null;
  private location: Location = { latitude: 0, longitude: 0 };
```


the difference is if you need some values passed as arguments of constructor, they will be available only in constructor. But in your case you could shrink amount of code by declaring values of properties directly, because no args passed in constructor.